src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -951,6 +951,11 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded, Comp
                                 }
                             }
                         }
+                        else
+                        {
+                            // We don't have any generators, so if we have a compilation from a previous run with generated files, we definitely can't use it anymore
+                            compilationWithStaleGeneratedTrees = null;
+                        }
```

Comment:
serious request.  please invert the `if` statement here.  having this case fall off hte end us so easy to miss.

Reply:
Good suggestion, will do.